build.go

```diff
@@ -271,22 +271,17 @@ func Build(projectPathParser PathParser,
 						return packit.BuildResult{}, err
 					}
 				}
-
 			} else {
 				logger.Process("Reusing cached layer %s", layer.Path)
 				if !build {
-					err := os.RemoveAll(filepath.Join(projectPath, "node_modules"))
-					if err != nil {
-						return packit.BuildResult{}, err
-					}
-
-					err = os.Symlink(filepath.Join(layer.Path, "node_modules"), filepath.Join(projectPath, "node_modules"))
+					err = ensureNodeModulesSymlink(projectPath, layer.Path, tmpDir)
 					if err != nil {
 						return packit.BuildResult{}, err
 					}
 				}
 			}
 
+			layer.Cache = true
```

Comment:
We shouldn't need to set the cache flag on a launch layer. In this case, I believe the issue is that we've moved the `layer.ExecD` assignment outside of the `if run` conditional block which means it gets set on every run. Then, without the cache flag set, the `exec.d` binary getting written to the layer wipes out the previous contents. If you move the `layer.ExecD` setting back to its original location, then you can remove this flag.

Reply:
@ryanmoran Thanks for the review. Are you sure about the original location, though? Originally, it was located at launch>run>build, but with that some caching test (vendored) fail. I moved it to launch>run and the test succeeds and I removed the `layer.Cache=true`. WDYT?